controller/backup_controller.go

```diff
@@ -324,6 +324,10 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 		if err != nil {
 			return
 		}
+
+		if bc.backupInFinalState(backup) && (!backup.Status.LastSyncedAt.IsZero() || backup.Spec.SnapshotName == "") {
```

Comment:
Good catch that we need to check `bc.backupInFinalState(backup)` before doing the clean up. Otherwise, we might cleanup the ticket and detach the volume before the backup CR is updated to be completed.

Reply:
Right - we did exactly that.